doc/releases/release-notes-4.0.rst

```diff
@@ -825,6 +825,20 @@ USB
 Devicetree
 **********
 
+* Added support for string-array and array type properties to be enums.
+  Many new macros added for this, for example :c:macro:`DT_ENUM_IDX_BY_IDX`.
+* Added :c:`DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY`.
+* Added :c:macro:`DT_NODE_HAS_STATUS_OKAY`.
+* Added :c:macro:`DT_INST_NUM_IRQS`.
+* Added macros :c:macro:`DT_NODE_FULL_NAME_UNQUOTED`, :c:macro:`DT_NODE_FULL_NAME_TOKEN`,
+  and :c:macro:`DT_NODE_FULL_NAME_UPPER_TOKEN`.
+* DT_*_REG_ADDR now returns an explicit unsigned value with C's ``U`` suffix.
+* The deprecated _ENUM_TOKEN and _ENUM_UPPER_TOKEN macros have been removed.
```

Comment:
the drop of the deprecated macros is already mentioned in "Removed APIs in this release", not sure we need both